### PR TITLE
Upgrade to ownCloud 9.0.1.

### LINKED ISSUE #54: Upgrade to ownCloud 9.0.1

### PR DESCRIPTION
**Pull request by kyrofa**
This PR fixes #54 by upgrading ownCloud to 9.0.1. It also starts using the .htaccess file shipped in the root of ownCloud. Note that this was done with a direct include rather than enabling overrides due to the performance penalty. Typically this would be unsafe, but the dangers are negated by the fact that the .htaccess file is read-only.

### CODE REVIEW

**Comment by kyrofa** [comment]
@oparoz would you mind taking a look at this when you're able?

File: src/owncloud/apache_config [hunk 1]
```diff
@@ -11,6 +11,20 @@ Alias "/apps" "${SNAP_DATA}/owncloud/apps"
...
+    # case an attacker was able to modify it, but that's not actually possible
+    # on Snappy (since the .htaccess file is read-only) so we'll do it here so
+    # as to avoid manually copying it in and needing to maintain it.
+    Include ${SNAP}/htdocs/.htaccess
+
+    # Increase the max upload size, and upload into a different tmp so we don't
+    # try to use that much RAM.
+    php_value upload_max_filesize 16G
+    php_value post_max_size 16G
+    php_value upload_tmp_dir ${SNAP_DATA}/owncloud/tmp
```
Comment (oparoz) [inline src/owncloud/apache_config:26]: Maybe also include the other PHP tmp variables?

This is what I use for PHP-FPM

```
; Temp folders
env[TMP] = ${SNAP_DATA}/var/tmp/
env[TMPDIR] = ${SNAP_DATA}/var/tmp/
env[TEMP] = ${SNAP_DATA}/var/tmp/
php_admin_value[upload_tmp_dir] = ${SNAP_DATA}/var/tmp/
```

File: src/owncloud/apache_config [hunk 2]
```diff
@@ -11,6 +11,20 @@ Alias "/apps" "${SNAP_DATA}/owncloud/apps"
...
-<IfModule dir_module>
-    DirectoryIndex index.html index.php
-</IfModule>
+<Directory "${SNAP}/htdocs">
+    # Include ownCloud's .htaccess file directly. In a typical setup this would
+    # be dangerous since it increases the capability of the .htaccess file in
+    # case an attacker was able to modify it, but that's not actually possible
+    # on Snappy (since the .htaccess file is read-only) so we'll do it here so
+    # as to avoid manually copying it in and needing to maintain it.
+    Include ${SNAP}/htdocs/.htaccess
```
Comment (oparoz) [inline src/owncloud/apache_config:20]: This still doesn't let users make changes to match their needs, but I guess this can be dealt with in another PR.

Comment (kyrofa) [inline src/owncloud/apache_config:20]: Indeed, that's not what this PR is about.

File: src/owncloud/apache_config (hunk 1 above)
Comment (kyrofa) [inline src/owncloud/apache_config:26]: The only reason I wanted to define the `upload_tmp_dir` is so we could upload files larger than the free space available in RAM. I don't see a problem with other things using `/tmp`. Do you?

Comment (oparoz) [inline src/owncloud/apache_config:26]: Well, given that some devices have very little RAM, I would prefer to avoid storing anything there, but maybe it's not needed until we can prove there is a problem. ownCloud does now have the possibility to specify the temp folder, but that doesn't mean that all apps are using it.

File: src/owncloud/apache_config (hunk 2 above)
Comment (oparoz) [inline src/owncloud/apache_config:20]: Don't you want to include the other `.htaccess` files though?

File: src/owncloud/apache_config (hunk 1 above)
Comment (kyrofa) [inline src/owncloud/apache_config:26]: Yeah I say leave it unless we know it's a problem. There's a performance hit with pinging the hard drive for this stuff.

File: src/owncloud/apache_config (hunk 2 above)
Comment (kyrofa) [inline src/owncloud/apache_config:20]: The only one I know about is in `config/`, and that folder is unused. Are there others?

**Comment by oparoz** [comment]
>The only one I know about is in config/, and that folder is unused.

Unused? We store the config in the config folder on the writeable partition. 

>Are there others?

There is on in data. It's the same as the one found in config.

File: src/owncloud/apache_config (hunk 2 above)
Comment (oparoz) [inline src/owncloud/apache_config:20]: > The only one I know about is in config/, and that folder is unused.

Unused? We store the config in the config folder on the writeable partition. 

> Are there others?

There is one in data. It's the same as the one found in config.

**Comment by kyrofa** [comment]
> Unused? We store the config in the config folder on the writeable partition.

Indeed, but that folder isn't reachable by URL (only the in-app `config/` folder is, which is unused).

> There is one in data. It's the same as the one found in config.

Yeah, same situation. Both of those are only useful if Apache directly serves those directories, which is doesn't. In fact, the only way I'd be able to include those `.htaccess` files is if I had `Directory` directives for them.

File: src/owncloud/apache_config (hunk 2 above)
Comment (kyrofa) [inline src/owncloud/apache_config:20]: > Unused? We store the config in the config folder on the writeable partition.

Indeed, but that folder isn't reachable by URL (only the in-app `config/` folder is, which is unused).

> There is one in data. It's the same as the one found in config.

Yeah, same situation. Both of those are only useful if Apache directly serves those directories, which it doesn't since both are outside `htdocs/`. In fact, the only way I'd be able to include those `.htaccess` files is if I had `Directory` directives for them. Might as well just leave them un-served, I say.

**Comment by oparoz** [comment]
I see what you mean. I remembered being able to read the files in /config, but it doesn't matter since they don't contain any sensitive information.

**Comment by kyrofa** [comment]
Alright, I'm gonna merge this, then.
